src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -1003,6 +1007,8 @@ InferenceProfiler::Measure(
     RETURN_IF_ERROR(manager_->GetAccumulatedClientStat(&start_stat));
   }
 
+  metrics_manager_->CheckQueryingStatus();
```

Comment:
Do we need to handle the error here?

Reply:
Good idea. Originally I was thinking of letting any exception propagate all the way up, but we should probably stick to error return objects here since they're currently used. Caught exception if there is one and returned error 👍